compiler/rustc_expand/src/mbe/macro_check.rs

```diff
@@ -558,8 +565,9 @@ fn check_ops_is_prefix(
     binders: &Binders,
     ops: &Stack<'_, KleeneToken>,
     span: Span,
-    name: MacroRulesNormalizedIdent,
+    name_opt: Option<MacroRulesNormalizedIdent>,
 ) {
+    let Some(name) = name_opt else { return };
```

Comment:
Optional because some meta-variable expressions do not have an attached `Ident` like `index` or `length`.

Reply:
It is also worth noting that 

https://github.com/rust-lang/rust/blob/f8a29bd95e0427ea1a46412ee2bb48e92644edc9/compiler/rustc_expand/src/mbe/transcribe.rs#L522

and

https://github.com/rust-lang/rust/blob/f8a29bd95e0427ea1a46412ee2bb48e92644edc9/compiler/rustc_expand/src/mbe/transcribe.rs#L531-L532

already check `Ident`s on the LHS and RHS